tests/integration/test_sns.py

```diff
@@ -1070,6 +1070,36 @@ def test_not_found_error_on_get_subscription_attributes(self):
         self.sns_client.delete_topic(TopicArn=topic_arn)
         self.sqs_client.delete_queue(QueueUrl=queue_url)
 
+    def test_message_to_fifo_sqs(self):
+        topic_name = "topic-{}.fifo".format(short_uid())
+        queue_name = "queue-%s.fifo" % short_uid()
+
+        topic_arn = self.sns_client.create_topic(Name=topic_name, Attributes={"FifoTopic": "true"})[
+            "TopicArn"
+        ]
+        queue = self.sqs_client.create_queue(
+            QueueName=queue_name,
+            Attributes={"FifoQueue": "true"},
+        )
+
+        queue_url = queue["QueueUrl"]
+        queue_arn = aws_stack.sqs_queue_arn(queue_name)
+
+        self.sns_client.subscribe(TopicArn=topic_arn, Protocol="sqs", Endpoint=queue_arn)
```

Comment:
Could we also add a negative test here, which exposes the new validations that have been added. Something like (after creating a standard, non-fifo queue):
```
with self.assertRaises(Exception) as ctx:
    self.sns_client.subscribe(TopicArn=topic_arn, Protocol="sqs", Endpoint=standard_queue_arn)
self.assertIn("FIFO SQS Queues can not be subscribed to standard SNS topics", str(ctx.exception))
```

(Btw, we have recently started migrating from `unittest` to `pytest` to define our test methods, fixtures, and assertions. With `pytest`, the block syntax would be `with pytest.raises(Exception) as e: ...`. However, we need to transform the entire file to `pytest` first before we can leverage this syntax..)